tests/test_main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+from unittest import mock
+
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_not_substantive(mocked_prediction):
+    mocked_prediction.return_value = 3
+
+    assert cryptocurrency_action(3) == "Do nothing"
+    assert cryptocurrency_action(3.1546) == "Do nothing"
+    assert cryptocurrency_action(2.8598) == "Do nothing"
```

Comment:
Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.